From 9c6f17aa2e2f90cfda5ec4bd0e5398156fa9a493 Mon Sep 17 00:00:00 2001
From: Catalin Horghidan <catalin.horghidan@nxp.com>
Date: Tue, 26 Mar 2019 15:27:29 +0200
Subject: [PATCH 675/706] ocelot: fdb: add new MAC entry for CPU port on
 PGID_CPU

Signed-off-by: Catalin Horghidan <catalin.horghidan@nxp.com>
(cherry picked from commit fea3afb1046520b9cb063f5f7e2b8d90b304dced)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/ocelot.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/mscc/ocelot.c b/drivers/net/ethernet/mscc/ocelot.c
index b047861bb162..e90082100d63 100644
--- a/drivers/net/ethernet/mscc/ocelot.c
+++ b/drivers/net/ethernet/mscc/ocelot.c
@@ -726,6 +726,7 @@ static int ocelot_fdb_add(struct ndmsg *ndm, struct nlattr *tb[],
 {
 	struct ocelot_port *port = netdev_priv(dev);
 	struct ocelot *ocelot = port->ocelot;
+	int swp = port->chip_port;
 
 	if (!vid) {
 		if (!port->vlan_aware)
@@ -741,8 +742,9 @@ static int ocelot_fdb_add(struct ndmsg *ndm, struct nlattr *tb[],
 			return -EINVAL;
 	}
 
-	return ocelot_mact_learn(ocelot, port->chip_port, addr, vid,
-				 ENTRYTYPE_LOCKED);
+	return ocelot_mact_learn(ocelot,
+				 ocelot->cpu_port_id == swp ? PGID_CPU : swp,
+				 addr, vid, ENTRYTYPE_LOCKED);
 }
 
 static int ocelot_fdb_del(struct ndmsg *ndm, struct nlattr *tb[],
-- 
2.17.1

